File: wxDesigner/samples2.19/dynamic/dynamic.py
#!/bin/env python
#----------------------------------------------------------------------------
# Name:         dynamic.py
# Author:       XXXX
# Created:      XX/XX/XX
# Copyright:    
#----------------------------------------------------------------------------

import wx

from dynamic_wdr import *


# WDR: classes

class MyDialog(wx.Dialog):
    def __init__(self, parent, id, title,
        pos = wx.DefaultPosition, size = wx.DefaultSize,
        style = wx.DEFAULT_DIALOG_STYLE ):
        wx.Dialog.__init__(self, parent, id, title, pos, size, style|wx.RESIZE_BORDER)
        
        # WDR: dialog function MyDialogFunc for MyDialog
        MyDialogFunc(self, True)
        
        self.sub_page = MyPageOneFunc( self, False, False )
        self.my_sizer.Add( self.sub_page )
        
        self.GetSizer().Fit( self )
        
        self.CentreOnParent()
        
        # WDR: handler declarations for MyDialog
        wx.EVT_CHOICE(self, ID_CHOICE, self.OnPage)

    # WDR: methods for MyDialog

    # WDR: handler implementations for MyDialog

    def OnPage(self, event):
        self.sub_page.DeleteWindows()
        self.my_sizer.Remove( self.sub_page )
        if event.GetInt() == 1:
            self.sub_page = MyPageTwoFunc( self, False, False )
        else:
            self.sub_page = MyPageOneFunc( self, False, False )
        self.my_sizer.Add( self.sub_page )
        self.GetSizer().Fit( self )

class MyFrame(wx.Frame):
    def __init__(self, parent, id, title,
        pos = wx.DefaultPosition, size = wx.DefaultSize,
        style = wx.DEFAULT_FRAME_STYLE ):
        wx.Frame.__init__(self, parent, id, title, pos, size, style)
        
        self.CreateMyMenuBar()
        
        self.CreateStatusBar(1)
        self.SetStatusText("Welcome!")
        
        # insert main window here
        
        # WDR: handler declarations for MyFrame
        wx.EVT_MENU(self, ID_TEST, self.OnTest)
        wx.EVT_MENU(self, wx.ID_ABOUT, self.OnAbout)
        wx.EVT_MENU(self, wx.ID_EXIT, self.OnQuit)
        wx.EVT_CLOSE(self, self.OnCloseWindow)
        
    # WDR: methods for MyFrame
    
    def CreateMyMenuBar(self):
        self.SetMenuBar( MyMenuBarFunc() )
    
    # WDR: handler implementations for MyFrame
    
    def OnTest(self, event):
        dialog = MyDialog( self, -1, "Test dialog" )
        dialog.ShowModal()
        dialog.Destroy()

    def OnAbout(self, event):
        dialog = wx.MessageDialog(self, "Welcome to SuperApp 1.0\n(C)opyright Joe Hacker",
            "About SuperApp", wx.OK|wx.ICON_INFORMATION )
        dialog.CentreOnParent()
        dialog.ShowModal()
        dialog.Destroy()
    
    def OnQuit(self, event):
        self.Close(True)
    
    def OnCloseWindow(self, event):
        self.Destroy()
    

#----------------------------------------------------------------------------

class MyApp(wx.App):
    
    def OnInit(self):
        frame = MyFrame(None, -1, "SuperApp", [20,20], [500,340])
        frame.Show(True)
        return True

#----------------------------------------------------------------------------

app = MyApp(True)
app.MainLoop()

File: wxDesigner/samples2.19/dynamic/dynamic.pl
#!/usr/bin/perl
#----------------------------------------------------------------------------
# Name:         dynamic.pl
# Author:       XXXX
# Created:      XX/XX/XX
# Copyright:    
#----------------------------------------------------------------------------

use strict;
use Wx;

do 'dynamic_wdr.pl';

# constants

# WDR: classes

package MyDialog;

use strict;
use vars qw(@ISA);

@ISA=qw(Wx::Dialog);

use Wx qw(wxID_OK wxID_CANCEL wxALL);

sub new {
    my( $class ) = shift;
    my( $this ) = $class->SUPER::new( @_ );
    
    # WDR: dialog function MyDialogFunc for MyDialog
    $this->SetSizer( &main::MyDialogFunc( $this, 1 ) );

    $this->{sub_page} = &main::MyPageOneFunc( $this, 0, 0 );
    $this->{my_sizer}->Add( $this->{sub_page} );
    $this->GetSizer()->Fit( $this );
    
    $this->CentreOnParent();
    
    # WDR: handler declarations for MyDialog
    Wx::Event::EVT_CHOICE( $this, $main::ID_CHOICE, \&OnPage );
    
    $this;
}

# WDR: methods for MyDialog

# WDR: handler implementations for MyDialog

sub OnPage {
    my( $this, $event ) = @_;
    $this->{sub_page}->DeleteWindows();
    $this->{my_sizer}->RemoveSizer( $this->{sub_page} );
    if( $event->GetInt() == 1 ) {
        $this->{sub_page} = &main::MyPageTwoFunc( $this, 0, 0 );
    } else {
        $this->{sub_page} = &main::MyPageOneFunc( $this, 0, 0 );
    }
    $this->{my_sizer}->Add( $this->{sub_page} );
    $this->GetSizer()->Fit( $this );
}

package MyFrame;

use strict;
use vars qw(@ISA);

@ISA=qw(Wx::Frame);

use Wx::Event qw(EVT_MENU EVT_CLOSE EVT_SIZE EVT_UPDATE_UI);
use Wx qw(wxOK wxICON_INFORMATION wxTB_HORIZONTAL wxNO_BORDER wxID_EXIT wxID_ABOUT);
use Wx qw(wxDefaultPosition wxDefaultSize wxDEFAULT_DIALOG_STYLE wxRESIZE_BORDER );

sub new {
    my( $class ) = shift;
    my( $this ) = $class->SUPER::new( @_ );

    $this->CreateMyMenuBar();
    
    $this->CreateStatusBar( 1 );
    $this->SetStatusText( "Welcome!", 0);
    
    # insert main window here
    
    # WDR: handler declarations for MyFrame
    EVT_MENU( $this, $main::ID_TEST, \&OnTest );
    EVT_MENU( $this, wxID_ABOUT, \&OnAbout );
    EVT_MENU( $this, wxID_EXIT, \&OnQuit );
    EVT_CLOSE( $this, \&OnCloseWindow );
    
    $this;
}

# WDR: methods for MyFrame

sub CreateMyMenuBar {
    my( $this ) = shift;
    
    $this->SetMenuBar( &main::MyMenuBarFunc() );
}

# WDR: handler implementations for MyFrame

sub OnTest {
    my( $this, $event ) = @_;
    
    my( $dialog ) = MyDialog->new( $this, -1, 'Test dialog', 
        wxDefaultPosition, wxDefaultSize, wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER );
    $dialog->ShowModal();
}

sub OnAbout {
    my( $this, $event ) = @_;
    
    Wx::MessageBox( "Welcome to SuperApp 1.0\n(C)opyright Joe Hacker",
        "About SuperApp", wxOK|wxICON_INFORMATION, $this );
}

sub OnQuit {
    my( $this, $event ) = @_;
    
    $this->Close(1);
}

sub OnCloseWindow {
    my( $this, $event ) = @_;
    
    $this->Destroy();
}

package MyApp;

use strict;
use vars qw(@ISA);

@ISA=qw(Wx::App);

sub OnInit {
    my( $this ) = @_;
    
    my( $frame ) = MyFrame->new( undef, -1, "SuperApp", [20,20], [500,340] );
    $frame->Show(1);
    
    1;
}

package main;

my( $app ) = MyApp->new();
$app->MainLoop();

